hack/testing/rsyslog/roles/rsyslog/tasks/main.yaml

```diff
@@ -0,0 +1,159 @@
+
+# Back up the pre-existing rsyslog config files in the backup dir,
+# then removing the files/dirs in rsyslog.d.
+- name: Moving the contents of {{rsyslog__config_dir}} to the backup dir
+  shell: |
+    tar cf - /etc/rsyslog.conf {{rsyslog__config_dir}} | (cd "{{ tempdir }}"; tar xf -)
+    rm -rf {{rsyslog__config_dir}}/*
+
+- name: create rsyslog viaq subdir
+  file: path="{{ rsyslog__viaq_path }}" state=directory mode=0700
+  when: rsyslog__viaq|bool
+
+- name: Update directory and file permissions
+  shell: |
+    [ ! -d {{ rsyslog__systemlog_dir }} ] || ( [ "$(stat -c '%G' {{ rsyslog__systemlog_dir }})" = "{{ rsyslog__group }}" ] || ( chown -v root:{{ rsyslog__group }} {{ rsyslog__systemlog_dir }} ; chmod -v 775 {{ rsyslog__systemlog_dir }} ) )
+  register: rsyslog__register_file_permissions
+  when: rsyslog__unprivileged|bool
+  changed_when: rsyslog__register_file_permissions.stdout != ''
+  notify: [ 'Restart rsyslogd' ]
+
+- name: Generate main rsyslog configuration
+  template:
+    src: 'etc/rsyslog.conf.j2'
+    dest: '/etc/rsyslog.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0400'
+  notify: [ 'Restart rsyslogd' ]
+  when: rsyslog__enabled|bool
+
+- name: Generate rsyslog configuration rules
+  template:
+    src: 'etc/rsyslog.d/rules.conf.j2'
+    dest: '{{rsyslog__config_dir}}/{{ item.filename | d((item.weight if item.weight|d() else rsyslog__weight_map[item.type|d("rules")]) + "-" + (item.name|d("rules")) + "." + (item.suffix |d ("conf"))) }}'
+    owner: '{{ item.owner | d("root") }}'
+    group: '{{ item.group | d("root") }}'
+    mode:  '{{ item.mode  | d("0400") }}'
+  with_flattened:
+    - '{{ rsyslog_pools | d([]) }}'
+    - '{{ rsyslog__default_rules }}'
+    - '{{ rsyslog__rules }}'
+    - '{{ rsyslog__group_rules }}'
+    - '{{ rsyslog__host_rules }}'
+    - '{{ rsyslog__dependent_rules }}'
+  when: (rsyslog__enabled|bool and rsyslog__example|bool and (item.filename|d() or item.name|d()) and
+         (item.state is undefined or item.state != 'absent') and
+         (item.options|d() or item.sections|d()))
+  notify: [ 'Restart rsyslogd' ]
+
+# rsyslog__default_viaq_rules:
+- name: Generate rsyslog configuration rules
+  template:
+    src: 'etc/rsyslog.d/rules.conf.j2'
+    dest: '{{rsyslog__config_dir}}/{{ item.filename | d((item.weight if item.weight|d() else rsyslog__weight_map[item.type|d("rules")]) + "-" + (item.name|d("rules")) + "." + (item.suffix |d ("conf"))) }}'
+    owner: '{{ item.owner | d("root") }}'
+    group: '{{ item.group | d("root") }}'
+    mode:  '{{ item.mode  | d("0400") }}'
+  with_flattened:
+    - '{{ rsyslog__default_viaq_rules }}'
+  when: (rsyslog__enabled|bool and rsyslog__viaq|bool and (item.filename|d() or item.name|d()) and
+         (item.state is undefined or item.state != 'absent') and
+         (item.options|d() or item.sections|d()))
+  notify: [ 'Restart rsyslogd' ]
+
+# rsyslog__viaq_rules:
+- name: Generate rsyslog viaq configuration configuration files
```

Comment:
is having "configuration configuration" intentional?

Reply:
Ahhh, a good catch. Thanks!